spec/integeration/usershow_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.feature 'User show page', type: :feature do
+  # Create a user with associated data
+  let!(:user) { FactoryBot.create(:user, name: 'TestUser', bio: 'This is a test bio.') }
+
+  before do
+    # Create some posts for the user
+    FactoryBot.create_list(:post, 5, author: user)
+
+    visit user_path(user)
+  end
+
+  scenario 'User sees user details, bio, and posts' do
+    # Check if user's profile picture is visible (assuming you have an image tag)
+    expect(page).to have_css('img') # Adjust this selector as needed
+
+    # Check if user's username is visible
+    expect(page).to have_content(user.name)
+
+    # Check if user's bio is visible
+    expect(page).to have_content(user.bio)
+
+    # Check if the first 3 posts are visible
+    user.posts.first(3).each_with_index do |post, _index|
+      expect(page).to have_content(post.title)
+      expect(page).to have_content(post.text.truncate(50)) # Truncate post body for visibility
+    end
+
+    # Check if the "View All Posts" button is visible
+    expect(page).to have_link('See all posts', href: user_posts_path(user))
+  end
+
+  scenario 'User clicks on a user post and is redirected to its show page' do
+    # Get the user's first post
+    post = user.posts.first
+
+    # Visit the user's post show page directly
+    visit user_post_path(user, post)
+
+    # Check if the page is redirected to the post's show page
+    expect(page).to have_current_path(user_post_path(user, post))
+  end
```

Comment:
- [ ] Your test here claims to click the user's post and then redirect but that is not correct. You are visiting the user's post, instead of clicking the post. Kindly implement the click event and ensure the current_path matches what is expected.